stages/stage_1.py

```diff
@@ -48,14 +48,15 @@ def __init__(self):
         operator_lookup.update((k.encode("utf8"), v) for (k, v) in BINDINGS.items())
         self.operator_lookup = operator_lookup
 
-    def __call__(self, program, args, max_cost=None, pre_eval_f=None):
+    def __call__(self, program, args, operator_lookup=None, max_cost=None, pre_eval_f=None, strict=False):
         return run_program_0(
             program,
             args,
             operator_lookup=self.operator_lookup,
```

Comment:
This isn't respecting the `operator_lookup`. Also, note that there is a data structure which looks both like a function and a dictionary which is generally used here (dictionary of atoms to functions, plus a fallback function to deal with strict). You can completely deal with strict vs non-strict there.

Reply:
I think this comment was from the first version of this patch too, right?